website/docs/guides/version_4_upgrade.html.markdown

```diff
@@ -0,0 +1,144 @@
+
+## Which Terraform versions are supported?
+
+Terraform versions `0.12.x` and greater are fully supported. Support for `0.11.x` has been removed.
+If you are still on one of the `0.11.x` versions we recommend upgrading to the latest stable release of Terraform.
+
+Please see the [Terraform Upgrade Guide](https://www.terraform.io/upgrade-guides/index.html)
+for more info about upgrading Terraform.
+
+## I accidentally upgraded to 4.0.0, how do I downgrade to `2.X`?
+
+If you've inadvertently upgraded to `4.0.0`, first see the
+[Provider Version Configuration Guide](#provider-version-configuration) to lock
+your provider version; if you've constrained the provider to a lower version
+such as shown in the previous version example in that guide, Terraform will pull
+in a `2.X` series release on `terraform init`.
+
+If you've only run `terraform init` or `terraform plan`, your state will not
+have been modified and downgrading your provider is sufficient.
+
+If you've run `terraform refresh` or `terraform apply`, Terraform may have made
+state changes in the meantime.
+
+- If you're using a *local* state, `terraform refresh` with a downgraded
+  provider is likely sufficient to revert your state.
+- If you're using a *remote* state backend
+  - That does not support versioning, see the local state instructions above
+  - That supports *versioning* you can revert the Terraform state file to a previous
+    version by hand. If you do so and Terraform created resources as part of a
+    `terraform apply`, you'll need to either `terraform import` them or delete
+    them by hand.
+
+## Upgrade Topics
+
+<!-- TOC depthFrom:2 depthTo:2 -->
+
+- [Provider Version Configuration](#provider-version-configuration)
+- [Provider Policy Changes](#provider-policy-changes)
+
+<!-- /TOC -->
+
+## Provider Version Configuration
+
+-> Before upgrading to version `4.0.0`, it is recommended to upgrade to the most
+recent version of the provider (`3.25.0`) and ensure that your environment
+successfully runs [`terraform plan`](https://www.terraform.io/docs/commands/plan.html)
+without unexpected changes or deprecation notices.
+
+It is recommended to use [version constraints](https://www.terraform.io/docs/configuration/providers.html#provider-versions)
+when configuring Terraform providers. If you are following that recommendation,
+update the version constraints in your Terraform configuration and run
+[`terraform init`](https://www.terraform.io/docs/commands/init.html) to download
+the new version.
+
+If you aren't using version constraints, you can use `terraform init -upgrade`
+in order to upgrade your provider to the latest released version.
+
+For example, given this previous configuration:
+
+```hcl
+provider "vault" {
+  # ... other configuration ...
+
+  version = "~> 3.25.0"
+}
+```
+
+An updated configuration:
+
+```hcl
+provider "vault" {
+  # ... other configuration ...
+
+  version = "~> 4.0.0"
+}
+```
+
+## Provider Policy Changes
+
```

Comment:
Can we add a bit more detail what these policy changes mean (what does the arrow from one operation to another mean), and what action the user needs to take (e.g. update/add X to their token/role policy)?

Reply:
done